run/events-pubsub/Startup.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Text;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Newtonsoft.Json;
+
+namespace EventsPubSub
+{
+    public class Startup
+    {
+        public void ConfigureServices(IServiceCollection services)
+        {
+        }
+
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
+        {
+            if (env.IsDevelopment())
+            {
+                app.UseDeveloperExceptionPage();
+            }
+
+            app.UseRouting();
+
+            app.UseEndpoints(endpoints =>
+            {
+                // [START run_events_pubsub_handler]
+                endpoints.MapPost("/", async context =>
+                {
+                    using (var reader = new StreamReader(context.Request.Body))
+                    {
+                        var body = await reader.ReadToEndAsync();
+                        dynamic cloudEventData = JsonConvert.DeserializeObject(body);

```

Comment:
@jskeet It will be possible to use the event serialization packages that you are working on on Cloud Run as well, I suppose? So when that's ready some of this code can be simplified?

Reply:
A note on this. I could use CloudEvents SDK to read the event and also possibly Jon's libraries in Functions Frameworks to read the PubSub envelope. However, the Python and Node.js samples deliberately not use any external library to read the events, rather rely on standard libraries. So, I just followed that pattern here as well.